backend/pipeline_readme.md

```diff
@@ -0,0 +1,303 @@
+
+## Installing/Adding a Service
+
+To add a service worker, as an admin, navigate to admin/service workers (url: `/admin/services`). Click the "Create New Service Worker" button and specify a name for the worker, as well as the configuration. The configuration will be a JSON body, typically provided by the service itself. Generally this will contain details on how to contact the service. See [Below](#web-version-1) for the web configuration schema as an example of what to place here. Once the name and configuration have been specified, click "Create", and processing of _new evidence_ should begin. Old evidence can be processed on an item-by-item basis, or in a batched manner on an operation-by-operation basis.
+
+## Building a Compliant Service
+
+There are various options in how you can construct a valid service. The primary concern is how to safely contact this service. Once contacted, the service is then responsible for the following:
+
+* Determining if the evidence is appropriate to analyze
+* Responding with the result
+
+The exact manner in which the above is accomplished is largely up to the service, so long as it can be configured using a standard configuration template. The below sections specify how to create a configuration hosted by various means.
+
+### Before You Start / Definitions
+
+The details below try to provide all of the necessary details to get you started in building a new pipeline service. To that end, the documentation here assumes you know a little bit about the following:
+
+* What AShirt is, and broadly what it can do
+* JSON format, and building/parsing JSON in the language of your choice
+* Basic Typescript type definitions. Specifically, you should have an understanding of:
+  * Basic javascript types (`string`, `number`, `undefined`, etc)
+  * Literal values (`'a'` means literally `'a'` in the message)
+  * Optional types (a key with a `?` suffix means this key/value can be omitted)
+  * Union types (`a | b` means either `a` or `b`)
+  * Record types (objects, or key/value maps. Key and values are restricted to the indicated types)
+  * A cheatsheet exists [here](https://www.typescriptlang.org/static/TypeScript%20Types-4cbf7b9d45dc0ec8d18c6c7a0c516114.png)
+
+Additional details may be necessary for the particular type of service you implment.
+
+### Web Service
+
+Web services are any service that can be contacted via an HTTP request, and respond in kind. The details on exactly how the request is sent and what the body is are below. Note that this can include AWS Lambda services, though there is a more specific tool available if you want to use AWS Lambda.
+
+#### Web Configurations
+
+Web services, like all pipeline services, must define their configuration to AShirt when adding the service. The configuration provided below is mostly concerned with how to contact your service, and provides some ways for ashirt to minimally customize the message your service receives.
+
+##### Web, Version 1
```

Comment:
Can we include an example config for the lambda service here also? (unless it is covered elsewhere)